### ISSUE TITLE
Inference mode vs training mode

### ISSUE THREAD

**Issue by guiandrade2** [openai/improved-gan#49]
I don't know if this question belongs here, but I am currently making a custom tf keras gan with feature matching loss and I am struggling to understand when to use inference mode on a model, that is, making use of training layers like dropout and updating batch norm parameters. This goes both for discriminator and generator as I understand that they should be trained separately.
